omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,145 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_8;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.webservices.helper.TaskAction;
+import org.openmrs.module.webservices.rest.web.MockTaskServiceWrapper;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskActionResource1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskDefinitionResource1_8;
+import org.openmrs.scheduler.TaskDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Arrays;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+
+public class TaskActionController1_8Test extends MainResourceControllerTest {
+	
+	@Autowired
+	RestService restService;
+	
+	private TaskDefinition testTask = new TaskDefinition(1, "TestTask", "TestTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private TaskDefinition tempTask = new TaskDefinition(3, "TempTask", "TempTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private MockTaskServiceWrapper mockTaskServiceWrapper = new MockTaskServiceWrapper();
+	
+	@Before
+	public void setUp() throws Exception {
+		mockTaskServiceWrapper.registeredTasks.addAll(Arrays.asList(testTask, tempTask));
+		
+		TaskActionResource1_8 taskActionResource = (TaskActionResource1_8) restService
+		        .getResourceBySupportedClass(TaskAction.class);
+		taskActionResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	public void shouldScheduleTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+	}
+	
+	@Test
+	public void shouldShutdownTask() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask);
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
```

Comment:
How how are you that the task wasn't already shutdown even before the shutdown command?

Reply:
Here I am checking the Scheduled list before making shout down the request. The scheduled list will contain this task If it is not shut down. 
After the request, It will be removed from the Scheduled list (shutdown). Then I am checking for removal.

Reply:
Oh which line number are you checking that the task is not already shutdown?

Reply:
Ohh, Sorry really missed that :cry: 
I will add it quickly